codegen/generate.go

```diff
@@ -138,7 +138,7 @@ func WriteToFile(terragruntOptions *options.TerragruntOptions, basePath string,
 	contentsToWrite := fmt.Sprintf("%s%s", prefix, config.Contents)
 
 	const ownerWriteGlobalReadPerms = 0644
-	if err := os.WriteFile(targetPath, []byte(contentsToWrite), ownerWriteGlobalReadPerms); err != nil {
+	if err := os.WriteFile(targetPath, hclwrite.Format([]byte(contentsToWrite)), ownerWriteGlobalReadPerms); err != nil {
```

Comment:
@wakeful , this needs to be opt-in. The `generate` block can write arbitrary content, not just HCL content.

We could have a `fmt` attribute that determines whether to format the file, and users can set that to format the file.

We can also look to leverage the filename extension to have a smarter default for formatting (e.g. if the file has extension `hcl`, `tf` or `tofu`, default to format the file if not defined.